x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -141,12 +141,22 @@ export const validatePackagePolicy = (
     if (inputVars.length) {
       inputValidationResults.vars = inputVars.reduce((results, [name, configEntry]) => {
         results[name] = input.enabled
-          ? validatePackagePolicyConfig(
-              configEntry,
-              inputVarDefsByPolicyTemplateAndType[inputKey][name],
-              name,
-              safeLoadYaml
-            )
+          ? inputVarDefsByPolicyTemplateAndType[inputKey] === undefined
+            ? [
+                i18n.translate('xpack.fleet.packagePolicyValidation.missingInputKeyMessage', {
+                  defaultMessage: '{inputKey} has no vars in policy template',
```

Comment:
I added validation error here, though I don't like it too much, it would be better to include this logic in `validatePackagePolicyConfig`.
I'm thinking the best place to add the validation error is when `varDef` is not defined here, meaning all unrecognized variables would be rejected with validation error. What do you think?
https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/services/validate_package_policy.ts#L214

Reply:
Yes it would be better 👍

Reply:
done